### ISSUE TITLE
Look at potential Android/iOS support

### ISSUE THREAD

**Issue by CurryKitten** [CurryKitten/CurryKitten-Sim#10]
Some potential users have asked if a tablet version is possible.  It is possible, but would currently involve some reengineering to make it more friendly for mobile devices.  Also, I can't publish to a non-jailbroken iOS device without paying money to be an official apple developer and then jump through many hoops to get it onto the app store.  At least I can just publish an apk file on Android.  Marking this as a future item - I'll look at it once I'm happy with everything on the other platforms

**Comment by CurryKitten** [comment]
As a proof of concept, I worked on compiling over to Android and used a 2015 old phone I had to try the sim.  

![IMG_7588](https://user-images.githubusercontent.com/12443664/75668799-c71f5280-5c71-11ea-8918-13405bc942b7.jpg)

Whilst it technically worked, I didn't have a USB cable to connect up a radio, so don't know if that side worked, but was able to connect a Blutooth keyboard so I could at least start the cars off.  This resulted in the sim running at around 10fps max.  I'm not sure if this is because I'm using an old phone, or that everything was designed around a desktop... or perhaps a combination of both.

Whilst it's encouraging getting the compilation to work, it seems we'd at least have to put in some effort to optimize things quite heavily for mobile release.

**Comment by CurryKitten** [comment]
I worked on getting an iPhone version of 047b running after learning you could deploy apps to test on your won phone without a paid developer license.  That said it wasn't as simple as Android and needed a great deal of hoop jumping.  But as my iPhone is relatively up to date (an iPhone XR) I thought it would at least give me an insight into how viable this was.

The iPhone doesn't allow OTG connections like Android, but you can have Bluetooth connections, so I looked up a PS4 controller and Bluetooth keyboard and ran it that way.  It actually ran pretty well - given that this was the desktop code and no optimizations had been done.  I made a short video here 

https://youtu.be/TKnloXb7Trw

We'd certainly need to expand the touch interface so the keyboard wasn't needed.  A quick poll of people suggests that although a mobile (i.e phone) version might not be wanted, people might like to use it with a controller on a tablet.  So will give it more attention after everything else.

**Comment by CurryKitten** [comment]
I'm now actively working on this to produce a 0.55b version of the sim for mobile.  As part of the development of the city level (and its subsequent optimizations), I wanted to see how this would run on a mobile device and so ended up adding some touch controls.  I found that it ran nicely on an iPhone XR, but very badly on a cheapo £150 unbranded tablet from Amazon.

The reason for this is simply the difference in the graphical and CPU power.  I found using a 3dmark benchmark like slingshot useful.  The iPhone gets 4500 and the unbranded tablet was getting 139.  So we need something at least into the 1000's to make it work.

Sadly though, I do have an issue with iOS and the City level in that the textures simply don't work and we get all black buildings.  I'm having a call with the asset developer to see if we can figure out the problem.  I don't know if the Android version is free from this issue, I've just ordered a tablet to help with my development/testing.
